buildkitd/dockerd-wrapper.sh

```diff
@@ -106,32 +106,54 @@ EOF
         fi
         if [ "$fail" = "true" ]; then
             # Print dockerd logs on start failure.
-            echo "==== Begin dockerd logs ===="
-            cat /var/log/docker.log
-            echo "==== End dockerd logs ===="
+            print_dockerd_logs
             echo "If you are having trouble running docker, try using the official earthly/dind image instead"
             return 1
         fi
         i=$((i+1))
     done
 }
 
+print_dockerd_logs() {
+  echo "==== Begin dockerd logs ===="
+  cat /var/log/docker.log
+  echo "==== End dockerd logs ===="
+}
+
 stop_dockerd() {
     dockerd_pid="$(cat /var/run/docker.pid)"
+
+    # Kill dockerd in a subshell and wait for the PID to exit. This ensures that dockerd cannot keep any files in
+    # EARTHLY_DOCKERD_DATA_ROOT open.
+    (kill_dockerd)
+    wait "$dockerd_pid" || true
```

Comment:
can we move this lint into `kill_dockerd` (as well as the `dockerd_pid="$(cat /var/run/docker.pid)"`).

and then just change it to call and block on `kill_dockerd` (rather than `(kill_dockerd)`).

Then there's no need to have `dockerd_pid` set in this function.